vortex-dtype/src/ptype.rs

```diff
@@ -105,6 +105,21 @@ macro_rules! match_each_integer_ptype {
     })
 }
 
+#[macro_export]
+macro_rules! match_each_float_ptype {
+    ($self:expr, | $_:tt $enc:ident | $($body:tt)*) => ({
+        macro_rules! __with__ {( $_ $enc:ident ) => ( $($body)* )}
+        use $crate::PType;
+        use vortex_dtype::half::f16;
```

Comment:
you might also want/need `use half::f16` in case it's not imported in places this macro is used

Reply:
The vortex_dtype::half is effectively a "vendored" version of half. This ensures we use consistent thing everywhere and also makes the implementation somewhat internal.  As in, we could replace this for a different f16 crate.